Plutarch/Api/V1/AssocMap.hs

```diff
@@ -47,3 +122,339 @@ instance
       x' <- Plutus.fromData x
       y' <- Plutus.fromData y
       Just (x', y')
+
+-- | Tests whether the map is empty.
+pnull :: Term s (PMap k v :--> PBool)
+pnull = plam (\map -> List.pnull # pto map)
+
+-- | Look up the given key in a 'PMap'.
+plookup :: (PIsData k, PIsData v) => Term (s :: S) (k :--> PMap k v :--> PMaybe v)
+plookup = phoistAcyclic $
+  plam $ \key ->
+    plookupDataWith
+      # (phoistAcyclic $ plam $ \pair -> pcon $ PJust $ pfromData $ psndBuiltin # pair)
+      # pdata key
+
+-- | Look up the given key data in a 'PMap'.
+plookupData :: (PIsData k, PIsData v) => Term (s :: S) (PAsData k :--> PMap k v :--> PMaybe (PAsData v))
```

Comment:
Please don't add these unnecessary kind signatures at the use sites in definition types.
`forall (s :: S).` is OK, but `A (s :: S)` isn't at the use-site. It's essentially the same as doing `f (x :: X)` unnecessarily at the term-level.